Source/JavaScriptCore/dfg/DFGOSREntry.cpp

```diff
@@ -213,13 +212,16 @@ void* prepareOSREntry(VM& vm, CallFrame* callFrame, CodeBlock* codeBlock, Byteco
             }
             value = jsDoubleNumber(value.asNumber());
             format = FlushedDouble;
+        } else {
+            if (value.isDouble() && abstractValue.isType(SpecInt32Only)) {
+                if (value.isInt32AsAnyInt())
+                    value = jsNumber(value.asInt32AsAnyInt());
```

Comment:
Should we bail out for this `else` case since we expected a Int32 but got non-Int32 double?

Reply:
Changed.